filters/filter.go

```diff
@@ -130,35 +198,17 @@ func (f tagsFilter) String() string {
 	return buf.String()
 }
 
-func (f tagsFilter) Matches(id string) bool {
-	if len(f.filters) == 0 {
-		return true
-	}
-	iter := f.iterFn(id)
-	defer iter.Close()
-
-	currIdx := 0
-	for iter.Next() && currIdx < len(f.filters) {
-		name, value := iter.Current()
-		if name < f.filters[currIdx].name {
-			continue
-		}
-		// If the current filter tag doesn't exist, bail immediately
-		if name > f.filters[currIdx].name {
+func (f *multiFilter) Matches(val string) bool {
+	for _, filter := range f.filters {
+		match := filter.Matches(val)
+		if f.op == Conjunction && match == false {
 			return false
 		}
-		// If the current filter value doesn't match provided value, bail immediately
-		if !f.filters[currIdx].filter.Matches(value) {
-			return false
+
+		if f.op == Disjunction && match == true {
+			return true
 		}
-		currIdx++
 	}
-	return iter.Err() == nil && currIdx == len(f.filters)
-}
 
-// TODO(xichen): add support for
-// * more sophisticated pattern-based filters (e.g., starts-with, any, contains, not, etc.)
-// * combined filters (e.g., and filters, or filters, etc.)
-func newFilter(id string) IDFilter {
-	return newEqualityFilter(id)
+	return f.op == Conjunction
```

Comment:
Yeah good point - agreed it should be consistent. Will check and return for true.